In [1]:
import numpy as np
import statistics
from scipy import stats

In [2]:
#Problem 1
list = [6, 7, 5, 7, 7, 8, 7, 6, 9, 7, 4, 10, 6, 8, 8, 9, 5, 6, 4, 8]
mean = np.mean(list)
median = np.median(list)
mode = stats.mode(list)
std_dev = statistics.stdev(list)
print("Mean is :", mean)
print("Median is :", median)
print("Mode is :", mode)
print("Standard Deviation is :", std_dev)

Mean is : 6.85
Median is : 7.0
Mode is : ModeResult(mode=array([7]), count=array([5]))
Standard Deviation is : 1.6311119875071343


In [3]:
#Problem 2
list2 = [28, 122, 217, 130, 120, 86, 80, 90, 140, 120, 70, 40, 145, 113, 90, 68, 174, 194, 170, 100, 75, 104, 97, 75,
123, 100, 75, 104, 97, 75, 123, 100, 89, 120, 109]
mean = np.mean(list2)
median = np.median(list2)
mode = stats.mode(list2)
std_dev = statistics.stdev(list2)
print("Mean is :", mean)
print("Median is :", median)
print("Mode is :", mode)
print("Standard Deviation is :", std_dev)

Mean is : 107.51428571428572
Median is : 100.0
Mode is : ModeResult(mode=array([75]), count=array([4]))
Standard Deviation is : 39.33892805484412


In [4]:
#Problem3
x = [0, 1, 2, 3, 4, 5]
fx = [0.09, 0.15, 0.40, 0.25, 0.10, 0.01]
mean = sum([fx[i]*x[i] for i in range(len(x))])
variance = sum([((mean-x[i])**2)*fx[i] for i in range(len(x))])
print("Mean is :", mean)
print("Variance is :",variance)

Mean is : 2.15
Variance is : 1.2275


In [5]:
from scipy.stats import binom
n=6
p =.3
q = 2
add = sum(np.random.binomial(n,p,50000) == q)/50000
print("Probability is :",add)
mean,var = binom.stats(n,p)
std = round(var**.5,2)
print("Mean is : ", mean)
print("Standard Deviation for this process is : ", std)

Probability is : 0.32606
Mean is :  1.7999999999999998
Standard Deviation for this process is :  1.12


Gaurav and Barakha are both preparing for entrance exams. Gaurav attempts to solve 8 questions per day with a correction rate of 75%, while Barakha averages around 12 questions per day with a correction rate of 45%. What is the probability
that each of them will solve 5 questions correctly? What happens in cases of 4 and 6 correct solutions? What do you infer from it? What are the two main governing factors affecting their ability to solve questions correctly? Give a pictorial
representation of the same to validate your answer.

In [6]:
#settings p and n values
n_gaurav = 8
p_gaurav = .75
n_barkha = 12
p_barkha = .45

#defining range for r
r_values_gaurav = range(n_gaurav+1)
r_values_barkha = range(n_barkha+1)

#list of pmf values
dist_gaurav = [binom.pmf(r,n_gaurav,p_gaurav) for r in r_values_gaurav]
dist_barkha = [binom.pmf(r,n_barkha,p_barkha) for r in r_values_barkha]

#plotting the graph
fig = plt.figure(figsize=(18,8))
plt.subplot(1,2,1)
plt.title('Gaurav\'s probability distribution')
plt.xlabel('Number of questions solved correctly')
plt.ylabel('Probability of solving')
plt.bar(r_values_gaurav,dist_gaurav)
plt.subplot(1,2,2)
plt.title('Barkha\'s probability distribution')
plt.xlabel('Number of questions solved correctly')
plt.ylabel('Probability of solving')
plt.bar(r_values_barkha,dist_barkha)
plt.show()


NameError: name 'plt' is not defined

Problem Statement 7:
Customers arrive at a rate of 72 per hour to my shop. What is the probability of 𝑘 customers arriving in 4 minutes? a) 5 customers, b) not more than 3 customers, c) more than 3 customers. Give a pictorial representation of the same to validate your answer.

In [ ]:
from scipy.stats import poisson
import seaborn as sb
data_binom = poisson.rvs(mu=4.8, size=200)

ax = sb.distplot(data_binom)
ax.set(xlabel='Customers per 4 minutes', ylabel='Frequency')
plt.title('Poisson distribution')
plt.show()

In [ ]:
import numpy as np
x = np.arange(0,25)
prob=poisson.cdf(x,4.8)
y = prob[3]
print('Probability of not more than 3 customers is', round(y,3))
print('Probability of more than 3 customers is', round(1-y,3))
plt.plot(prob)
plt.plot((3,3),(0,1))
plt.scatter(3,prob[3])
plt.text(3,prob[3],'({}, {})'.format(3, round(prob[3],3)))
plt.title('CDF of poisson distribution')
plt.show()

#Answer11
Mean current is 10mA, Variance is 4mA, Standard deviation is 2. Z score for 13 mA Current is (13-10)/2 = 1.5. 
We need to find P(Z>z) = P(Z>1.5) = 1-P(Z<1.5).
Using tables, we find that (P<1.5c) is .933. So prabability of P(Z>1.5) is 1-.933 = .067. 
Prabability that a current exceeding 13mA will flow is .33
Z-Score for 9 mA current is (9-10)/2 =-.5 
Z Score for 11mA current is (11-10)/2 = .5
We need to find P(-0.5<Z<0.5) = P(Z<0.5)-P(Z<-0.5) = 0.69-0.309 = 0.382 ~~ 0.38
P(Z<z) = 0.98 
We need to find z using tables, z comes around 2.05.
current equation of z is : 2.05*2+10 = 4.1+10 = 14.1mA